setup.py

```diff
@@ -1190,6 +1190,7 @@ def detect_readline_curses(self):
             self.add(Extension('_curses_panel', ['_curses_panel.c'],
                            include_dirs=curses_includes,
                            define_macros=curses_defines,
+                           extra_compile_args=['-DPy_BUILD_CORE_MODULE'],
```

Comment:
Why is it needed?

Reply:
The new function ``_PyType_DisabledNew`` is only available when the macro is defined. My PR GH-25713 unifies the definition of the macro.